### ISSUE TITLE
Coverage calculation issue

### ISSUE THREAD

**Issue by adongare** [projecttacoma/fqm-execution#281]
### Summary
There seems to be some issue with the overage calculation. This measure(bundles attached) is supposed to have 100% coverage but for some reason looks like a couple of clauses are not being included while calculating coverage which drops the coverage by one percentage.

#### External Tracking Ticket
https://oncprojectracking.healthit.gov/support/browse/BONNIEMAT-1617

### Expected Behavior
100% coverage

### Version or Commit
v1.3.1

### Inputs (e.g. Measure Bundle, Patient Bundle, CQL Library)
Measure bundle: [CMS108-measure.zip](https://github.com/projecttacoma/fqm-execution/files/12796787/CMS108-measure.zip)
Patient bundles:  [CMS108-patients.zip](https://github.com/projecttacoma/fqm-execution/files/12796789/CMS108-patients.zip)

### Relevant Calculation Options (e.g. Measurement Period, meta.profile Validation)
```
{
    includeClauseResults: false,
    trustMetaProfile: true,
    buildStatementLevelHTML: true,
    measurementPeriodStart: 1/1/2025,
    measurementPeriodEnd: 12/31/2025,
}
```
### cql-to-elm Version Used for Measure Logic Translation (if known)
v2.11.0

**Comment by hossenlopp** [comment]
Hi @adongare!

Thanks for adding all the data we need for looking into this. On a quick look, this may be an issue with the coverage highlighting and perhaps a small lack in test data that isn't apparent in the highlighting. Our team is starting a new sprint this week and will be prioritizing this issue.

**Comment by hossenlopp** [comment]
Hi @adongare,

It looks like there are some visual coverage issues in this case that were hiding three uncovered clauses. The coverage 
percentage calculation is correct and the user should be able to get to 100% by adding test cases to cover the following
cases: 

### One Clause in define "Intervention Comfort Measures":

The `InterventionRequest.doNotPerform` is not covered by the test cases.

```
define "Intervention Comfort Measures":
  ( ["ServiceRequest": code in "Comfort Measures"] InterventionRequest
      where InterventionRequest.intent in { 'order', 'original-order', 'reflex-order', 'filler-order', 'instance-order' }
      and InterventionRequest.status in { 'active', 'on-hold', 'completed'}
      and InterventionRequest.doNotPerform is not true
  )
    union ( ["Procedure": "Comfort Measures"] InterventionPerformed
        where InterventionPerformed.status in { 'completed', 'in-progress' }
    )
```

There is no test case that has a "Comfort Measure" ServiceRequest with `doNotPerform` set to `true`. Adding a test that
has this field set to `true` would cover the `InterventionRequest.doNotPerform` clause.


### Two Clauses in define "Encounter with Prior or Present Diagnosis of Atrial Fibrillation or Prior Diagnosis of VTE":

Two code comparisons are not being fulfilled in this define statement in the third part of the union that makes up the statement. `VTEDiagnosis.clinicalStatus ~ QICoreCommon."remission"` and `VTEDiagnosis.clinicalStatus ~ QICoreCommon."resolved"` clauses are not covered, specifically the (`~`) equivalence operators.

```
union ( VTE."Encounter with Age Range and without VTE Diagnosis or Obstetrical Conditions" QualifyingEncounter
    with ["Condition": "Venous Thromboembolism"] VTEDiagnosis
        such that ( VTEDiagnosis.clinicalStatus  ~ QICoreCommon."inactive"
            or VTEDiagnosis.clinicalStatus  ~ QICoreCommon."remission"
            or  VTEDiagnosis.clinicalStatus ~ QICoreCommon."resolved"
        )
        and VTEDiagnosis.verificationStatus is not null and VTEDiagnosis.verificationStatus ~ QICoreCommon."confirmed"
        and VTEDiagnosis.onset.toInterval() before start of QualifyingEncounter.period
)
```

The test cases only satisfy the `VTEDiagnosis.clinicalStatus ~ QICoreCommon."inactive"` case. The `"remission"` and 
`"resolved"` cases can be covered by adding test data that uses those codes for the `clinicalStatus` element.


## Coverage Highlighting Issues

While the user can get to 100% with adding test cases, there are still issues with the coverage highlighting that makes
it hard for them to have known where the uncovered clauses are. We are continuing to investigate and will give an update
when we have a fix.

Best,
Chris

**Comment by adongare** [comment]
Hi @hossenlopp, thank you so much for the thorough analysis. We'll go ahead and add the test cases for the above scenarios.

**Comment by hossenlopp** [comment]
Hi @adongare!

Thank you for attaching the additional measures and tests cases. We are continuing to look into these issues.

**Comment by jkotanchik-SB** [comment]
Hi @hossenlopp,

Another group of users are reporting coverage issues. Would you prefer I attach the materials to this issue, or open new issues?

There are 7 measures mentioned in the report and are all some combination of <100% coverage with unhighlighted aliases, union operators, and/or attributes (that are covered in one or more test cases).

**Comment by hossenlopp** [comment]
Hi @adongare,

The additional measures and test cases you posted on 10/11 don't look like normal test case exports since they are simply a list of patient bundles. We are able to run this format anyways but when we try to reproduce these issues locally, we are only seeing minimal coverage of around 10-20%. Is there some additional issue with these test cases we are not aware of?

**Comment by hossenlopp** [comment]
Hi @adongare,

We have just released [v1.3.2](https://github.com/projecttacoma/fqm-execution/releases/tag/v1.3.2) which fixes an issue with visual coverage and should now properly show locations in the logic that are uncovered in many cases. This update should make it clear where those last few uncovered clauses are and aid in creating additional test cases. _Note: This does not fix visual coverage issues with some alias, unions and library usages. These do not affect the coverage percentage calculation._

Please test with this release and open a new issue if there are additional problems with coverage or you are unable to get to 100% after satisfying the clauses that will now properly show as uncovered.
